warehouse/forklift/legacy.py

```diff
@@ -496,6 +517,23 @@ def _verify_url(url: str, publisher_url: str | None) -> bool:
     )
 
 
+def _verify_url(
+    url: str, publisher_url: str | None, project_name: str, project_normalized_name: str
+) -> bool:
+    is_verified_pypi = _verify_url_pypi(
+        url=url,
+        project_name=project_name,
+        project_normalized_name=project_normalized_name,
+    )
+
+    if publisher_url:
+        return is_verified_pypi or _verify_url_with_trusted_publisher(
+            url=url, publisher_url=publisher_url
+        )
+    else:
+        return is_verified_pypi
```

Comment:
Short-circuit, maybe?

(Not sure if this is actually better, but it avoids a few unnecessary fall-throughs 🙂)

```suggestion
    if _verify_url_pypi(
        url=url,
        project_name=project_name,
        project_normalized_name=project_normalized_name,
    ):
        return True
        
    if not publisher_url:
        return False

    return _verify_url_with_trusted_publisher(
        url=url, publisher_url=publisher_url
    )
```